dotCMS/src/main/java/com/dotcms/rest/BundleResource.java

```diff
@@ -625,6 +637,140 @@ public Response deleteAllSuccess(@Context final HttpServletRequest request,
                 "Removing bundles in a separated process, the result of the operation will be notified")).build();
     } // deleteAllSuccess.
 
+    
+    @Path("/_download/{bundleId}")
+    @GET
+    @JSONP
+    @NoCache
+    @Produces(MediaType.APPLICATION_OCTET_STREAM)
+    @Consumes(MediaType.APPLICATION_JSON)
+    public final Response downloadBundle(@Context final HttpServletRequest request,
+                                       @Context final HttpServletResponse response,
+                                       @PathParam("bundleId") final String bundleId) throws DotPublisherException {
+    
+        User user = new WebResource.InitBuilder(request, response)
+                        .requiredBackendUser(true)
+                        .rejectWhenNoUser(true)
+                        .requiredPortlet("publishing-queue")
+                        .init()
+                        .getUser();
+        
+
+        final Bundle bundle = Try.of(()->APILocator.getBundleAPI().getBundleById(bundleId)).getOrElseThrow(e->new DotRuntimeException(e));
+
+        File bundleFile = new File(  ConfigUtils.getBundlePath() + File.separator + File.separator + bundle.getId() + ".tar.gz" );
```

Comment:
Codacy found an issue: [The String literal ".tar.gz" appears 5 times in this file; the first occurrence is on line 661](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6239861)